tests/components/reolink/test_host.py

```diff
@@ -0,0 +1,83 @@
+"""Test the Reolink host."""
+
+from asyncio import CancelledError
+from unittest.mock import AsyncMock, MagicMock
+
+from aiohttp import ClientResponseError
+import pytest
+
+from homeassistant.components.reolink import const
+from homeassistant.components.webhook import async_handle_webhook
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.util.aiohttp import MockRequest
+
+from .conftest import TEST_UID
+
+from tests.common import MockConfigEntry
+from tests.typing import ClientSessionGenerator
+
+
+async def test_webhook_callback(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    config_entry: MockConfigEntry,
+    reolink_connect: MagicMock,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test webhook callback with motion sensor."""
+    assert await hass.config_entries.async_setup(config_entry.entry_id) is True
+    await hass.async_block_till_done()
+    assert config_entry.state is ConfigEntryState.LOADED
+
+    webhook_id = f"{const.DOMAIN}_{TEST_UID.replace(':', '')}_ONVIF"
+
+    signal_all = MagicMock()
+    signal_ch = MagicMock()
+    async_dispatcher_connect(hass, f"{webhook_id}_all", signal_all)
+    async_dispatcher_connect(hass, f"{webhook_id}_0", signal_ch)
+
+    client = await hass_client_no_auth()
+
+    # test webhook callback success all channels
+    reolink_connect.ONVIF_event_callback.return_value = None
+    await client.post(f"/api/webhook/{webhook_id}")
+    signal_all.assert_called_once()
+
+    # test webhook callback all channels with failure to read motion_state
+    reolink_connect.get_motion_state_all_ch.return_value = False
+    await client.post(f"/api/webhook/{webhook_id}")
+    signal_all.assert_called_once()
```

Comment:
Correct me if I am wrong, but shouldn't the mocks be reset in between usage?

Reply:
In this particular case not, since I am checking it is not fired again.
I could reset it and change to `assert_not_called` that is maybe more clear, will do that,

Reply:
Ah check, yea the current test looks like it assumes that its called after every call

Reply:
improved